include/boost/exception/info.hpp

```diff
@@ -175,6 +185,22 @@ boost
             return x;
             }
 
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+        template <class E,class Tag,class T>
+        inline
+        E const &
+        set_info( E const & x, error_info<Tag,T> && v )
+            {
+            typedef error_info<Tag,T> error_info_tag_t;
+            shared_ptr<error_info_tag_t> p( new error_info_tag_t(std::move(v)) );
```

Comment:
use `shared_ptr<error_info_tag_t> p = make_shared<error_info_tag_t>(std::move(v))`
It calls less memory allocations